trie/database.go

```diff
@@ -778,10 +778,15 @@ func (db *Database) Commit(node common.Hash, report bool, callback func(common.H
 // commit is the private locked version of Commit.
 func (db *Database) commit(hash common.Hash, batch ethdb.Batch, uncacher *cleaner, callback func(common.Hash)) error {
 	// If the node does not exist, it's a previously committed node
+
+	db.lock.RLock()
 	node, ok := db.dirties[hash]
 	if !ok {
```

Comment:
Isn't this the same as moving the `RUnlock` call right after the `db.dirties[hash]` read?

Reply:
Yes it is :)  thank you !